datafusion/core/src/physical_planner.rs

```diff
@@ -2135,7 +2135,15 @@ impl DefaultPhysicalPlanner {
             "Optimized physical plan:\n{}\n",
             displayable(new_plan.as_ref()).indent(false)
         );
-        debug!("Detailed optimized physical plan:\n{new_plan:?}");
+
+        // Don't print new_plan directly, as that may overflow the stack.
+        // For example:
+        // thread 'tokio-runtime-worker' has overflowed its stack
+        // fatal runtime error: stack overflow, aborting
+        debug!(
+            "Detailed optimized physical plan:\n{}\n",
+            displayable(new_plan.as_ref()).indent(true)
+        );
```

Comment:
Consistent with the logging at the top of this function 👍 

https://github.com/apache/datafusion/blob/deba0069ae5b3bc3a22d8dd3038718a566e98bb0/datafusion/core/src/physical_planner.rs#L2096-L2103